python/llm/src/ipex_llm/transformers/lookup.py

```diff
@@ -381,11 +388,16 @@ def lookup_generate(self,
             self.post_time.append(pot-mot)
 
         # Stop on eos and remove content after eos
-        output_ids_list = output_ids[0].tolist()
-        if generation_config.eos_token_id in output_ids_list:
-            idx = output_ids_list.index(generation_config.eos_token_id)
-            step -= (len(output_ids_list) - idx - 1)
-            break
+        if eos_token_id_set is not None:
+            output_ids_list = output_ids[0].tolist()
+            first_eos_idx = -1
+            for out_idx, out_id in enumerate(output_ids_list):
+                if out_id in eos_token_id_set:
+                    first_eos_idx = out_idx
+                    break
+            if first_eos_idx > -1:
+                step -= (len(output_ids_list) - first_eos_idx - 1)
+                break
```

Comment:
Is it better to do something like
```python
       for eos_token_id in eos_token_id_set
          if eos_token_id  in output_ids_list:
             idx = output_ids_list.index(eos_token_id)
             step -= (len(output_ids_list) - idx - 1)
             break
```

Reply:
We do this as we need to find the first token in `output_ids_list` which is in `eos_token_id` :)

Reply:
OK; then maybe
```python
      for out_idx, out_id in enumerate(output_ids_list):
           if out_id in eos_token_id_set:
                step -= (len(output_ids_list) - out_idx - 1)
                break
```

Reply:
Thank you for the comment :) But we need a flag here (i.e., `first_eos_idx `) to jump out of the token generation loop. By this way, we could only break the `output_ids_list` iteration loop, which may make eos token ineffective.

Reply:
ok